### PR TITLE
Fix(web): allow user to save event without making changes

### LINKED ISSUE #990: Cannot save event after making recurrence change
### Where did this happen?

Hosted (app.compasscalendar.com), Local code with an up-to-date main branch

### Expected Behavior

When editing existing event, can always save, even if user only changed a recurrence field

### Current Behavior

When changing a recurrence field of an existing event, user cannot save. Save button is disabled

### Steps to Reproduce
**Empty title flow**
1. Click grid to start a new draft
2. Form will appear
3. Don't change any fields (including title)
4. Click Save button
5. Expected: Event is saved with title of `""`
6. Actual: Cannot save

**Recurrence flow**
1. Click on an existing event
2. Click the recurrence field
3. Change a recurrence field
4. Expected: Save is enabled
5. Actual: Save is still disabled

### Possible Solution

- `isDirty` isn't working for one or more of the recurrence fields
- Remove the `isDirty` flow and instead rely on defaults when the form is open. This would allow the user to never run into the scenario where the save button is disabled

### Context

Validate this is working by writing at that an event's form and adjusts a recurrence field and then clicks the save button. The form should close and the new data should be reflected in the calendar

### PR DESCRIPTION
**Pull request by tyler-dane**
Follow up to #991, minus the Draft schema changes

Closes #990

----

This pull request refactors and improves the event "dirty check" logic in the web calendar application. It introduces a dedicated parser for determining if an event has been modified, adds comprehensive tests for this logic, and updates the event draft submission flow to use the new implementation. The changes also remove legacy or duplicated dirty-checking code, resulting in a cleaner and more robust codebase.

**Event Dirty Check Refactor and Testing**

- Introduced a new `WebEventParser` class in `event.parser.ts` to encapsulate the logic for determining if a web event draft has been modified ("dirty"), including nuanced handling of recurrence and date changes. Also exported a standalone `isEventDirty` function for convenience.
- Added comprehensive unit tests for the dirty check logic in `event.parser.test.ts`, covering all relevant event fields, recurrence scenarios, and edge cases.
- Created a utility function `createWebEvent` for generating test events in `test.event.util.ts` to support the new tests.

**Integration and Cleanup in Calendar Draft Actions**

- Updated `useDraftActions.ts` to use the new `isEventDirty` parser, removing the previous inline implementation and ensuring all dirty checks are consistent and robust. The draft submission logic now uses a new `determineSubmitAction` helper to decide whether to submit, discard, or just open the form based on event state. [[1]](diffhunk://#diff-e051b1a707546d00b0f7957cc69f844c58c9025f103cf430132e72a86da60556R18) [[2]](diffhunk://#diff-e051b1a707546d00b0f7957cc69f844c58c9025f103cf430132e72a86da60556L158-L184) [[3]](diffhunk://#diff-e051b1a707546d00b0f7957cc69f844c58c9025f103cf430132e72a86da60556R239-R274) [[4]](diffhunk://#diff-e051b1a707546d00b0f7957cc69f844c58c9025f103cf430132e72a86da60556R288-R290)
- Removed the legacy and now-redundant dirty check and new event identification tests from `useDraftActions.test.ts`.

**UI and Miscellaneous**

- Cleaned up the `GridDraft` component by removing unused code related to the old dirty check and save button disabling. [[1]](diffhunk://#diff-70839fe5c37810033daadcc2ddde3a03c8565e21848490b6215e4c5b14d026feL66-L70) [[2]](diffhunk://#diff-70839fe5c37810033daadcc2ddde3a03c8565e21848490b6215e4c5b14d026feL118)
- Improved error handling by ensuring errors are logged to the console for debugging.

### CODE REVIEW

File: packages/web/src/common/parsers/event.parser.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        ? this.isRecurrenceChanged()
+        : current !== original;
+    });
+  }
+
+  /**
+   * Private method to check if recurrence has changed
+   */
+  private isRecurrenceChanged(): boolean {
+    if (!this.orig) return false;
```
[the remark this answers is not in the document]
Comment (tyler-dane) [reply packages/web/src/common/parsers/event.parser.ts]: fixed

File: packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts [hunk 2]
```diff
@@ -262,24 +236,44 @@ export const useDraftActions = (
...
+          return;
+        case "DISCARD":
           discard();
-        }
-        return;
+          return;
+        case "SUBMIT":
+        default:
+          // Continue with the submit logic below
+          break;
```
[the remark this answers is not in the document]
Comment (tyler-dane) [reply packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts:274]: fixed

File: packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts [hunk 3]
```diff
@@ -262,24 +236,44 @@ export const useDraftActions = (
...
+  const determineSubmitAction = useCallback(
+    (draft: Schema_WebEvent) => {
+      const isExisting = draft._id;
+      const isOptimistic = draft._id?.startsWith(ID_OPTIMISTIC_PREFIX);
+      if (isExisting && !isOptimistic) {
+        if (isFormOpenBeforeDragging) {
+          openForm();
+          return "OPEN_FORM";
+        }
+        const isSame = reduxDraft ? !isEventDirty(draft, reduxDraft) : false;
```
[the remark this answers is not in the document]
Comment (tyler-dane) [reply packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts:247]: fixed

File: packages/web/src/common/parsers/event.parser.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tyler-dane) [reply packages/web/src/common/parsers/event.parser.ts]: fixed

File: packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts [hunk 4]
```diff
@@ -293,8 +285,9 @@ export const useDraftActions = (
       }
 
       const { startOfView, endOfView } = weekProps.component;
+      const isExisting = draft._id;
 
-      if (!isNewEvent) {
+      if (isExisting) {
```
[the remark this answers is not in the document]
Comment (tyler-dane) [reply packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts:291]: fixed

File: packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts [hunk 5]
```diff
@@ -262,24 +236,42 @@ export const useDraftActions = (
...
+        }
+        const isSame = reduxDraft ? !isEventDirty(draft, reduxDraft) : false;
+        if (isSame) {
+          // no need to make HTTP request
+          return "DISCARD";
+        }
+      }
+      return "SUBMIT";
+    },
+    [reduxDraft, isFormOpenBeforeDragging, openForm, discard],
```
[the remark this answers is not in the document]
Comment (tyler-dane) [reply packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts]: fixed

File: packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (tyler-dane) [reply packages/web/src/views/Calendar/components/Draft/hooks/actions/useDraftActions.ts:291]: fixed
